proto/v1/cel-service.proto

```diff
@@ -0,0 +1,120 @@
+syntax = "proto3";
+
+package google.api.expr.v1;
+
+option cc_enable_arenas = true;
+option java_multiple_files = true;
+option java_outer_classname = "CelServiceProto";
+option java_package = "com.google.api.expr.v1";
+
+import "google/rpc/status.proto";
+import "proto/checked/v1/checked.proto";
+import "proto/v1/eval.proto";
+import "proto/v1/syntax.proto";
+
+// Access a CEL implementation from another process or machine.
+// A CEL implementation is decomposed as a parser, a static checker,
+// and an evaluator.  Every CEL implementation is expected to provide
+// a server for this API.  The API will be used for conformance testing,
+// utilities, and execution as a service.
+service CelService {
+  // Transforms CEL source text into a parsed representation.
+  rpc Parse (ParseRequest) returns (ParseResponse);
+
+  // Runs static checks on a parsed CEL representation and return
+  // an annotated representation, or a set of issues.
+  rpc Check (CheckRequest) returns (CheckResponse);
+
+  // Evaluates a parsed or annotation CEL representation given
+  // values of external bindings.
+  rpc Eval (EvalRequest) returns (EvalResponse);
+}
+
+// Warnings or errors in service execution are represented by
+// google.rpc.Status messages, with the following message
+// in the details field.
+message StatusDetails {
+  enum Severity {
+    // An unspecified severity.
+    SEVERITY_UNSPECIFIED = 0;
+    // Deprecation issue for statements and method that may no longer be
+    // supported or maintained.
+    DEPRECATION = 1;
+    // Warnings such as: unused variables.
+    WARNING = 2;
+    // Errors such as: unmatched curly braces or variable redefinition.
+    ERROR = 3;
+  }
+
+  Severity severity = 1;
+
+  // The following fields specify location, if known.
+
+  // The 1-based index of the starting line in the source text
+  // where the issue occurs, or 0 if unknown.
+  int32 line = 2;
+  // The 0-based index of the starting position within the line of source text
+  // where the issue occurs.  Only meaningful if line is nonzer.
+  int32 column = 3;
+  // Expression ID from [syntax.proto], 0 if unknown.
+  int64 id = 4;
+}
+
+message ParseRequest {
+  // Source text in CEL syntax.
```

Comment:
The `source` and `location` probably belong in their own message since it may be annotated with other things like `title` and `description`. See [google.type.Expr](https://cloud.google.com/service-management/reference/rpc/google.type#google.type.Expr). Or rather, you could just use that value as input to your service. I think the `syntax` and `disable_macros` fields should remain though.

Reply:
We currently wouldn't do anything with a title or description field, since they have no place in ParsedExpr or its subcomponents.  I don't think that having an overlap of two string fields is sufficient justification for such a dependency.  Moreover, we'll need to rethink this all if CEL gets modules with sources from multiple locations - I'd rather keep the two simple string fields here for easier refactoring.